Firmware/ultralcd.cpp

```diff
@@ -6929,29 +6941,61 @@ bool lcd_selftest()
 
 	_progress = lcd_selftest_screen(testScreen::extruderFan, _progress, 3, true, 2000);
 #if (defined(FANCHECK) && defined(TACH_0))
-	_result = lcd_selftest_fan_dialog(0);
+	switch (lcd_selftest_fan_auto(0)){		// check extruder Fan
+		case TestError::extruderFan:
+			_result = false;
+			break;
+		case TestError::swappedFan:
+			_swapped_fan = true;
+		case TestError::noerror:
+			_result = true;
+			break;
+	}
 #else //defined(TACH_0)
 	_result = lcd_selftest_manual_fan_check(0, false);
+#endif //defined(TACH_0)
 	if (!_result)
 	{
 		lcd_selftest_error(TestError::extruderFan, "", "");
 	}
-#endif //defined(TACH_0)
-	
 
 	if (_result)
 	{
 		_progress = lcd_selftest_screen(testScreen::printFan, _progress, 3, true, 2000);
-#if (defined(FANCHECK) && defined(TACH_1)) 		
-		_result = lcd_selftest_fan_dialog(1);
+#if (defined(FANCHECK) && defined(TACH_1))
+	switch (lcd_selftest_fan_auto(1)){		// check print fan
```

Comment:
Bunch of compiler warnings are generated. You can suppress them by adding  default: or by using enum class having only three possible values as return value of this function.

sketch/ultralcd.cpp:6966:9: warning: enumeration value 'heater' not handled in switch [-Wswitch]
  switch (lcd_selftest_fan_auto(1)){  // check print fan
         ^
sketch/ultralcd.cpp:6966:9: warning: enumeration value 'bed' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6966:9: warning: enumeration value 'endstops' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6966:9: warning: enumeration value 'motor' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6966:9: warning: enumeration value 'endstop' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6966:9: warning: enumeration value 'extruderFan' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6966:9: warning: enumeration value 'pulley' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6966:9: warning: enumeration value 'axis' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6966:9: warning: enumeration value 'wiringFsensor' not handled in switch [-Wswitch]
sketch/ultralcd.cpp:6966:9: warning: enumeration value 'triggeringFsensor' not handled in switch [-Wswitch]